src/App.tsx

```diff
@@ -15,49 +17,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  none = '',
+  alphabet = 'alphabet',
+  length = 'length',
+}
+
+const GoodList = ({ goods }: { goods: string[] }) => (
+  <ul>
+    {goods.map(good => (
+      <li key={good} data-cy="Good">{good}</li>))}
+  </ul>
+);
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: { sortField: string, isReversed: boolean },
+) {
+  const prepearedGoods = [...goods];
+
+  if (sortField) {
+    prepearedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.alphabet:
+          return good1.localeCompare(good2);
+        case SortType.length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepearedGoods.reverse();
+  }
+
+  return prepearedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.none);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods
+  = getPreparedGoods(goodsFromServer, { sortField, isReversed });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info',
+            { 'is-light': sortField !== SortType.alphabet })}
+          onClick={() => {
+            setSortField(SortType.alphabet);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success',
+            { 'is-light': sortField !== SortType.length })}
+          onClick={() => setSortField(SortType.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning',
+            { 'is-light': !isReversed })}
+          onClick={() => setIsReversed(!isReversed)}
```

Comment:
use "functional" setState in such cases: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state